lib/io.c

```diff
@@ -139,3 +139,15 @@ int metal_io_block_set(struct metal_io_region *io, unsigned long offset,
 	return retlen;
 }
 
+bool metal_io_is_block_valid(struct metal_io_region *io, unsigned long offset,
+			     const void *addr, int len)
+{
+	if (!io || !addr)
+		return false;
+
+	if ((addr >= (io->virt + offset)) && ((addr + len) < (io->virt + io->size)))
```

Comment:
I missed that, you increment void pointers. you have to cast them before.
Could you also add a header to describe the parameters?
As this commit need update could you also add the offerflow test pointed out by @Ed ?

Reply:
@arnopo Yes, no problem - thank you for the suggestion.  Do you prefer uint8_t* or unsigned char*

I see virt cast to uint8_t in io.h but represented as unsigned char* in io.c

Reply:
uint8_t seems more reliable  but both are ok